### ISSUE TITLE
Diffs are different depending on length of text node value

### ISSUE THREAD

**Issue by joedj** [Shoobx/xmldiff#22]
I've noticed that diffs are different depending on the length of a text node.  Not sure whether this is intentional/configurable or not.

E.g.
if `len(old) == 8 and len(new) == 36` then result is a single `UpdateTextIn` diff
if `len(old) == 7 and len(new) == 36` then result is `InsertNode` + `UpdateTextIn` + `DeleteNode`

Tested on 2.0b7.

```python
#!/usr/bin/env python

import xmldiff.main
import unittest

def xml(length):
    return '<tag><inner>%s</inner></tag>' % ('X' * length,)

class Test(unittest.TestCase):

    def testSimilarLengths(self):
        old = xml(8)
        new = xml(36)
        diff = xmldiff.main.diff_texts(old, new)
        self.assertEquals(len(diff), 1, diff)

    def testDissimilarLengths(self):
        old = xml(7)
        new = xml(36)
        diff = xmldiff.main.diff_texts(old, new)
        self.assertEquals(len(diff), 1, diff)

if __name__ == '__main__':
    unittest.main()
```

Output:
```
⇒  ./test_length.py
F.
======================================================================
FAIL: testDissimilarLengths (__main__.Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test_length.py", line 21, in testDissimilarLengths
    self.assertEquals(len(diff), 1, diff)
AssertionError: [InsertNode(target='/tag[1]', tag='inner', position=0), UpdateTextIn(node='/tag/inner[1]', text='XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'), DeleteNode(node='/tag/inner[2]')]

----------------------------------------------------------------------
Ran 2 tests in 0.001s

FAILED (failures=1)
```

**Comment by regebro** [comment]
Yeah, if the texts are very dissimilar the matching algorithm will not see it as the same node, and you get an insert if a new node and a delete of the old. When they contain only text that's unnecessary, of course, but very hard to do anything about.

**Comment by regebro** [comment]
Hmmm. Perhaps if we spilt the matching up into more separate bits, like matching on attributes and matching on texts and matching on paths, there could be parameters/arguments to choose the weights for each bit, and if the XML always has the same structure, make paths very important, for example...

**Comment by strichter** [comment]
I think that it is correct default behavior to have different output based on text length. The weight of the text is important after all. I would be okay with a parameter to completely ignore text length. But that has to be explicit and not be the default.

**Comment by regebro** [comment]
During the optimizations that I did I looked at a few ways of modifying relative weights of different differences, but you end up having infinite abilities to tweak the matching, that only changes the outcome but doesn't actually make it significantly better.

Ignoring the length of the texts is only possibly if we ignore text similarity, a flag for that certainly is possible, then nodes would match if texts are exactly the same, and not match if they are not. That's only something you want if the XML file contains no human text, but only fields. A flag for that is certainly possible, but I'm not sure it actually has any usefulness.
